packages/react/src/components/form/Textfield/Textfield.tsx

```diff
@@ -1,217 +1,99 @@
-    const characterLimitId = `textfield-charactercount-${useId()}`;
-    const hasCharacterLimit = characterLimit != null;
-
-    const describedBy =
-      cl(
-        inputProps['aria-describedby'],
-        hasCharacterLimit && characterLimitId,
-      ) || undefined;
+    },
+    ref,
+  ) => {
+    const AffixWrapper = prefix || suffix ? FieldAffixWrapper : Fragment;
 
     return (
-      <Paragraph asChild data-size={size}>
-        <div
-          style={style}
-          className={cl(
-            `ds-textfield`,
-            `ds-textfield--${size}`,
-            readOnly && `ds-textfield--readonly`,
-            hasError && `ds-textfield--error`,
-            className,
-          )}
-        >
-          {label && (
-            <Label
-              data-size={size}
-              weight='medium'
-              htmlFor={inputProps.id}
-              className={cl(`ds-textfield__label`, hideLabel && `ds-sr-only`)}
-            >
-              {readOnly && (
-                <PadlockLockedFillIcon
-                  aria-hidden
-                  className='ds-textfield__readonly__icon'
-                />
-              )}
-              <span>{label}</span>
-            </Label>
-          )}
-          {description && (
-            <Paragraph asChild data-size={size}>
-              <div
-                id={descriptionId}
-                className={cl(
-                  `ds-textfield__description`,
-                  hideLabel && `ds-sr-only`,
-                )}
-              >
-                {description}
-              </div>
-            </Paragraph>
-          )}
-          <div className='ds-textfield__field'>
-            {prefix && (
-              <Paragraph asChild data-size={size} variant='short'>
-                <div
-                  className={cl(
-                    `ds-textfield__adornment`,
-                    `ds-textfield__prefix`,
-                  )}
-                  aria-hidden='true'
-                >
-                  {prefix}
-                </div>
-              </Paragraph>
-            )}
-            <input
-              className={cl(
-                `ds-textfield__input`,
-                `ds-focus`,
-                prefix && `ds-textfield__input--with-prefix`,
-                suffix && `ds-textfield__input--with-suffix`,
-              )}
-              ref={ref}
-              type={type}
-              disabled={inputProps.disabled}
-              aria-describedby={describedBy}
-              size={htmlSize}
-              {...omit(['size', 'error', 'errorId'], rest)}
-              {...inputProps}
-              onChange={(e) => {
-                inputProps?.onChange?.(e);
-                setInputValue(e.target.value);
-              }}
+      <Field data-size={size}>
+        {!!label && <Label>{label}</Label>}
+        {!!description && <FieldDescription>{description}</FieldDescription>}
+        <AffixWrapper>
+          {prefix && <FieldAffix>{prefix}</FieldAffix>}
```

Comment:
Skulle vi hatt
```suggestion
          {prefix === undefined || <FieldAffix>{prefix}</FieldAffix>}
```
slik at `prefix={0}` funker? ☺️

Reply:
fixed